test/gtest/ucm/malloc_hook.cc

```diff
@@ -241,8 +249,10 @@ void test_thread::test() {
     /* Release old pointers (should not crash) */
     old_ptrs.clear();
 
+    pthread_mutex_lock(&m_stats_lock);
```

Comment:
let's insert a barrier before calling this block, so that one thread will not remove the ranges before other threads check them